x-pack/platform/plugins/shared/streams/server/lib/telemetry/stats/streams_usage_collector.ts

```diff
@@ -0,0 +1,317 @@
+import { registerStreamsUsageCollector as registerCollector } from './register_collector';
+
+/**
+ * Creates a fetch function for Streams usage statistics with logger closure
+ */
+function createFetchFunction(logger: Logger) {
+  return async function fetchStreamsUsageStats({
+    esClient,
+  }: {
+    esClient: any;
+  }): Promise<StreamsStatsTelemetry> {
+    try {
+      const [allStreamsData, significantEventsMetrics, ruleExecutionMetrics] = await Promise.all([
+        fetchAllStreamsData(esClient),
+        fetchSignificantEventsMetrics(esClient),
+        fetchRuleExecutionMetrics(esClient),
+      ]);
+
+      const { classicStreamsMetrics, wiredStreamsCount, streamsCache } =
+        processStreamsData(allStreamsData);
+
+      // Pass the cache to avoid additional ES calls
+      const categorizedSignificantEventsMetrics = await categorizeSignificantEvents(
+        significantEventsMetrics,
+        streamsCache
+      );
+
+      return {
+        classic_streams: classicStreamsMetrics,
+        wired_streams: { count: wiredStreamsCount },
+        significant_events: {
+          ...categorizedSignificantEventsMetrics,
+          ...ruleExecutionMetrics,
+        },
+      };
+    } catch (error) {
+      logger.error('Failed to collect Streams telemetry data', error);
+      throw error;
+    }
+  };
+
+  async function fetchAllStreamsData(esClient: any) {
+    const allStreams = await esClient.search({
+      index: '.kibana_streams',
+      size: 10000,
+      sort: [{ name: 'asc' }],
+      track_total_hits: false,
+      _source: true,
+      query: { match_all: {} },
+    });
+
+    return allStreams.hits?.hits ?? [];
+  }
+
+  function processStreamsData(hits: any[]) {
+    let changedCount = 0;
+    let withProcessingCount = 0;
+    let withFieldsCount = 0;
+    let withChangedRetentionCount = 0;
+    let wiredCount = 0;
+
+    const streamsCache = new Map();
+
+    for (const hit of hits) {
+      const definition = hit._source ?? {};
+
+      if (isClassicStream(definition)) {
+        // Presence of a classic stream in `.kibana_streams` implies it has been stored/changed
+        changedCount++;
+
+        if (hasProcessingSteps(definition)) {
+          withProcessingCount++;
+        }
+
+        if (hasFieldOverrides(definition)) {
+          withFieldsCount++;
+        }
+
+        if (hasChangedRetention(definition.ingest?.lifecycle)) {
+          withChangedRetentionCount++;
+        }
+      }
+
+      if (isWiredStream(definition) && !isGroupStream(definition)) {
+        wiredCount++;
+      }
+
+      // Cache the stream definition for later use
+      streamsCache.set(definition.name, definition);
+    }
+
+    return {
+      classicStreamsMetrics: {
+        changed_count: changedCount,
+        with_processing_count: withProcessingCount,
+        with_fields_count: withFieldsCount,
+        with_changed_retention_count: withChangedRetentionCount,
+      },
```

Comment:
should we have a consistent naming for top level and nested property or is there a particular reason top level are in camel case ?